lib/msf/util/db_manager.rb

```diff
@@ -37,12 +37,20 @@ def self.process_opts_workspace(opts, framework, required = true)
       raise ArgumentError.new("opts must include a valid :workspace")
     end
 
-    if wspace.kind_of?(String)
+    case wspace
```

Comment:
Nice catch here, this seems pretty useful.

Reply:
@jmartin-r7 this might solve some of the Pro issues you were running into as well

Reply:
This is one of the paths in the possible solutions set.  Definitely an improvement on detection.

There are other possible ramifications as this does not yet handle when the workspace name is `change` from under the existing object by another db connection.